indico/modules/events/contributions/clone.py

```diff
@@ -98,14 +104,48 @@ def run(self, new_event, cloners, shared_data):
         self._contrib_field_map = shared_data['contribution_fields']['contrib_field_map']
         self._contrib_map = {}
         self._subcontrib_map = {}
+
         with db.session.no_autoflush:
-            self._clone_contribs(new_event)
+            if new_event == self.old_event:
+                self._clone_contrib()
+            else:
+                self._clone_contribs(new_event)
+
         self._synchronize_friendly_id(new_event)
         db.session.flush()
         return {'contrib_map': self._contrib_map, 'subcontrib_map': self._subcontrib_map}
 
-    def _clone_contribs(self, new_event):
+    def _create_new_contribution(self, event, contribution, excluded_attrs=None):
```

Comment:
I think keeping it named `old_contrib` is clearer, since then you have `<new|old>_contrib` instead of `new_contrib` and `contribution`.